gprofiler/py_package_version.py

```diff
@@ -0,0 +1,160 @@
+"""Package name and version for Python files.
+
+Some of the functions in this module are implemented based on similar functions
+in pip: _get_metadata, _convert_legacy_entry, _files_from_record,
+_files_from_legacy, _get_package_name.
+"""
+import csv
+import email
+import os
+import pathlib
+from typing import Iterator, List, Optional, Tuple
+
+import pkg_resources
+
+# A little monkey patch to prevent pkg_resources from converting "/proc/{pid}/root/" to "/"
+pkg_resources._normalize_cached = lambda path: path
+
+
+__all__ = ["get_versions"]
+
+
+def _convert_to_proc_root_path(path: str, pid: int) -> str:
+    assert path.startswith("/")
+    return os.path.join(f"/proc/{pid}/root", path)
+
+
+def _get_packages_dir(file_path: str) -> Optional[str]:
+    if not file_path.startswith("/"):
```

Comment:
why? `co_filename` can be relative.

Reply:
Not sure that's true, do you have any example?

Anyway I think what @YishaiZinkin tests here is to avoid stuff like `<string>` etc. Could be commented.

Reply:
Generally this module converts paths to `/proc/{pid}/root/` paths, so paths here should be absolute. I think it's pretty clear from the code, and described in the docstring of the main function.